src/Equations/damaged-elastic/Kernels/Time.cpp

```diff
@@ -703,3 +700,1491 @@ void seissol::kernels::Time::flopsTaylorExpansion(long long& nonZeroFlops,
+            m_nonlinearInterpolation;
+
+        m_nonLinInter.QInterpolated = &QInterpolatedPlus[timeInterval][0];
+        m_nonLinInter.Q = degreesOfFreedomPlus;
+        m_nonLinInter.execute(side, 0);
+
+        m_nonLinInter.QInterpolated = &QInterpolatedMinus[timeInterval][0];
+        m_nonLinInter.Q = degreesOfFreedomMinus;
+        m_nonLinInter.execute(cellInformation[l_cell].faceRelations[side][0],
+                              cellInformation[l_cell].faceRelations[side][1] + 1);
+      }
+
+      // S3: Construct matrices to store Rusanov flux on surface quadrature nodes.
+      // Reshape the interpolated results
+      using QInterpolatedShapeT =
+          const real(*)[seissol::dr::misc::numQuantities][seissol::dr::misc::numPaddedPoints];
+
+      auto* qIPlus = (reinterpret_cast<QInterpolatedShapeT>(QInterpolatedPlus));
+      auto* qIMinus = (reinterpret_cast<QInterpolatedShapeT>(QInterpolatedMinus));
+
+      // The arrays to store time integrated flux
+      alignas(PAGESIZE_STACK) real rusanovFluxPlus[tensor::QInterpolated::size()] = {0.0};
+      // alignas(PAGESIZE_STACK) real rusanovFluxMinus[tensor::QInterpolated::size()] = {0.0};
+
+      for (unsigned i_f = 0; i_f < tensor::QInterpolated::size(); i_f++) {
+        rusanovFluxPlus[i_f] = static_cast<real>(0.0);
+      }
+
+      using rusanovFluxShape = real(*)[seissol::dr::misc::numPaddedPoints];
+      auto* rusanovFluxP = reinterpret_cast<rusanovFluxShape>(rusanovFluxPlus);
+
+      // S4: Compute the Rusanov flux
+      computeNonLinearRusanovFlux(materialData,
+                                  l_cell,
+                                  side,
+                                  timeWeights,
+                                  *qIPlus[0],
+                                  *qIMinus[0],
+                                  *rusanovFluxP,
+                                  localIntegration);
+
+      /// S5: Integrate in space using quadrature.
+      kernel::nonlinearSurfaceIntegral m_surfIntegral = m_nonlSurfIntPrototype;
+      m_surfIntegral.Q = data.dofs;
+      m_surfIntegral.Flux = rusanovFluxPlus;
+      m_surfIntegral.fluxScale = localIntegration[l_cell].fluxScales[side];
+      m_surfIntegral.execute(side, 0);
+    } else if (cellInformation[l_cell].faceTypes[side] == FaceType::dynamicRupture) {
+      // No neighboring cell contribution, interior bc.
+      assert(reinterpret_cast<uintptr_t>(drMapping[l_cell][side].godunov) % ALIGNMENT == 0);
+
+      kernel::nonlinearSurfaceIntegral m_drIntegral = m_nonlSurfIntPrototype;
+      m_drIntegral.Q = data.dofs;
+      m_drIntegral.Flux = drMapping[l_cell][side].godunov;
+      m_drIntegral.fluxScale = localIntegration[l_cell].fluxScales[side];
+      m_drIntegral.execute(side, drMapping[l_cell][side].faceRelation);
+    } // if (faceTypes)
+  }   // for (side)
+}
+
+void seissol::kernels::Time::stressToDofsDynamicRupture(real* dofsStressNPlus,
```

Comment:
As far as I can see, you can replace this by a function, which only operates on one DOF set and call it twice for `Plus` and `Minus`.